lms/djangoapps/support/static/support/jsx/single_support_form.jsx

```diff
@@ -183,8 +193,66 @@ class RenderForm extends React.Component {
     );
   }
 
+  async handleInputChange(event) {
+    event.preventDefault();
+    const queryString = event.target.value;
+    if (queryString.length > 2) {
+      let suggestions = [];
+      await fetch(`${this.learnerSupportCenterURL}/hc/api/internal/instant_search.json?query=${queryString}&locale=en-us`, { headers: {
+        'Content-Type': 'application/json',
+      }}).then((response) => {
+        suggestions = response.json.results;
+        this.setState({ suggestions });
+      });
+    } else {
+      this.setState({
+        suggestions: [],
+        activeSuggestion: 0,
+      });
+    }
+  }
+
+  onKeyDown(event) {
+    const { activeSuggestion, suggestions } = this.state;
+
+    if (event.keyCode === 13) { // User pressed the enter key
+      window.location.href = this.learnerSupportCenterURL + suggestions[activeSuggestion].url;
+    } else if (event.keyCode === 38) { // User pressed the up arrow
+      if (activeSuggestion === 0) { return; }
+
+      this.setState({ activeSuggestion: activeSuggestion - 1 });
+    } else if (event.keyCode === 40) { // User pressed the down arrow
+      if (activeSuggestion - 1 === suggestions.length) { return; }
+
+      this.setState({ activeSuggestion: activeSuggestion + 1 });
+    }
+  }
+
+  handleBlur(event) {
+    if (!this._ignoreBlur) {
```

Comment:
what are the cases blur event will be ignored?

Reply:
When there are some results after we've entered the keywords in the input field, if I click outside input field, onBlur will invoke which sets suggestions to `[]` in the states. 
But when I try to click a suggestion, onBlur for input is invoked and the suggestions are gone before I can click on any of them. So in order to click a suggestion, I don't want to invoke onBlur.

Also, found out a fun and similar problem that someone had: https://stackoverflow.com/questions/39134574/delayed-onblur-callback